agent/agent.go

```diff
@@ -299,3 +330,11 @@ func (a *Agent) AddPeers(ctx context.Context, p pool.Pool, num int) error {
 type Service interface {
 	Whitelist(ctx context.Context, nodeID string) error
 }
+
+// enodeURItoID takes an enode:// URI string and returns the NodeID component.
+func enodeURItoID(uri string) string {
+	if len(uri) <= 8+128 {
+		return uri
```

Comment:
Yea, actually I don't think I even need this since ActivePeers is just IDs. I think I was using this helper for something else but it got factored out.

Reply:
Actually just realized that this might not work at all since ActivePeers is just IDs. Hmm, need to rethink the approach.

Reply:
I removed it altogether for now, so we're only comparing enodeIDs.

Might be good enough to launch this feature in v2.3 but I'd really like to get the host comparisons in so we can use it to replace any accidentally-externally-routed peers with proper internally-routed peers.